portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+let i = 0;
+let facts = [
+  "Digital Arts and Sciences Engineer ",
+  "Full-Stack Developer ",
+  "Scrum Master ",
+  "Google SPS participant! >:) "
+];
+let txt = "I'm a " + facts[0];
+let speed = 50;
+let backspace = false;
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+window.onload = function() {
+  typeWriter();
+  i = 0;
+  txt = "I'm a " + facts[0];
+  backspace = false;
+};
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function typeButton() {
+  typeWriter();
+  i = 0;
+  txt = "I'm a " + facts[Math.floor(Math.random() * facts.length)];
+  backspace = false;
+}
+
+function typeWriter() {
+  if (i <= txt.length && !backspace) {
+    document.getElementById("typed").innerHTML = txt.substring(0, i) + "|";
+    i++;
+    if (i === txt.length + 1) {
+      backspace = true;
+      setTimeout(wait(), 15000);
+    }
+    setTimeout(typeWriter, speed);
+  }
+  if (backspace) {
+    if (i >= 0) {
+      document.getElementById("typed").innerHTML = txt.substring(0, i) + "|";
+      i--;
+      setTimeout(typeWriter, speed + 10);
+    }
+  }
+}
+
+function wait() {
```

Comment:
Can you explain what this function does? I think it creates a blinking cursor, is that right? What does the `for` loop part do?

Reply:
Before I added that function, the typing was getting erased way too fast so it didn't give the viewer the opportunity to read the whole message. So, I tried making another setTimeout for that, but I needed to pass a function to it so that's why wait() exists. It's just making the cursor "blink" for a certain amount of time before erasing. If you have any suggestions on how to do all of this more efficiently, please let me know!

Reply:
Ohh interesting. I think using `setTimeout()` makes perfect sense, but my guess is that you don't need a `for` loop here. Maybe you could create a global boolean variable that tracks whether to show the `|` character, and then toggle that in a function that you call every .5 seconds or so?